### PR TITLE
Update environment and component versions.

### PR DESCRIPTION
**Pull request by nick863**
Update automl-gpu environment and pin new versions of components to automl-gpu v2.

### CODE REVIEW

**Review by vlbejan — DISMISSED** [review]

File: assets/training/forecasting_demand/environments/automl-gpu/context/conda_dependencies.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ dependencies:
 - openssl=1.1.1s
 - pip:
   - inference-schema
-  - cryptography==41.0.0
+  - cryptography==41.0.2
```
Comment (vizhur) [inline assets/training/forecasting_demand/environments/automl-gpu/context/conda_dependencies.yaml:25]: use 41.0.3 (Ideally you should remove this unless any of your dependency is pipping it that is unlikely)

Comment (nick863) [reply assets/training/forecasting_demand/environments/automl-gpu/context/conda_dependencies.yaml]: Fixed.

**Review by vlbejan — DISMISSED** [review]

File: assets/training/forecasting_demand/environments/automl-gpu/context/conda_dependencies.yaml [hunk 2]
```diff
@@ -5,24 +5,23 @@ channels:
...
+- GitPython=3.1.32
 - scipy~=1.7.0
 - tqdm
 - setuptools=65.5.1
 - wheel=0.38.1
 - openssl=1.1.1s
 - pip:
   - inference-schema
-  - cryptography==41.0.0
+  - cryptography==41.0.3
```
Comment (vizhur) [inline assets/training/forecasting_demand/environments/automl-gpu/context/conda_dependencies.yaml:24]: I don't think you need to have cryptography here

**Review by vizhur — DISMISSED** [review]

File: assets/training/forecasting_demand/environments/automl-gpu/context/conda_dependencies.yaml (hunk 2 above)
Comment (nick863) [reply assets/training/forecasting_demand/environments/automl-gpu/context/conda_dependencies.yaml]: Removed.

File: assets/training/forecasting_demand/environments/automl-gpu/context/conda_dependencies.yaml [hunk 3]
```diff
@@ -5,24 +5,22 @@ channels:
...
-- GitPython=3.1.30
+- GitPython=3.1.32
 - scipy~=1.7.0
 - tqdm
 - setuptools=65.5.1
 - wheel=0.38.1
 - openssl=1.1.1s
 - pip:
   - inference-schema
-  - cryptography==41.0.0
```
Comment (vlbejan) [inline assets/training/forecasting_demand/environments/automl-gpu/context/conda_dependencies.yaml:25]: Don't we have this package in the Forecasting DNN CE?

**Review by vlbejan — APPROVED** [review]

File: assets/training/forecasting_demand/environments/automl-gpu/context/conda_dependencies.yaml (hunk 3 above)
Comment (nick863) [reply assets/training/forecasting_demand/environments/automl-gpu/context/conda_dependencies.yaml:25]: Yes, but it seems, we do not require it.

**Review by vizhur — APPROVED** [review]

**Pull request merged**